phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -78,7 +87,11 @@
     private PhoenixConnection connection;
     private final long maxSize;
     private final ImmutableBytesPtr tempPtr = new ImmutableBytesPtr();
-    private final Map<TableRef, Map<ImmutableBytesPtr,Map<PColumn,byte[]>>> mutations = Maps.newHashMapWithExpectedSize(3); // TODO: Sizing?
+    /**
+     * Use {@link TreeMap} for {@link #mutations} below to enforce a deterministic ordering for easier testing of partial failures.
+     * @see PartialCommitIT
+     */
+    private final Map<TableRef, Map<ImmutableBytesPtr,RowMutationState>> mutations = Maps.newTreeMap(new TableRefComparator());
```

Comment:
I understand, but would it be workable to create a copy of the collection in the test and use Collections.sort() to sort it (since we don't care about perf in tests, as we do at runtime)?